packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,562 @@
+    Offset layoutOffset,
+  ) {
+    final Offset untransformedPaintingCoordinates = offset
+        + new Offset(
+            layoutOffset.dx,
+            _getUntransformedPaintingCoordinateY(layoutOffset.dy)
+        );
+
+    // Get child's center as a fraction of the viewport's height.
+    final double fractionalY =
+        (untransformedPaintingCoordinates.dy + _itemExtent / 2.0) / size.height;
+    final double angle = -(fractionalY - 0.5) * 2.0 * _maxVisibleRadian;
+    // Don't paint the backside of the cylinder when
+    // renderChildrenOutsideViewport is true.
+    if (angle > math.pi / 2.0 || angle < -math.pi / 2.0)
+      return;
+
+    final Matrix4 transform = MatrixUtils.createCylindricalProjectionTransform(
+      radius: size.height * _diameterRatio / 2.0,
+      angle: angle,
+      perspective: _perspective,
+    );
+
+    context.pushTransform(
+      needsCompositing,
+      offset,
+      _centerOriginTransform(transform),
+      // Pre-transform painting function.
+      (PaintingContext context, Offset offset) {
+        context.paintChild(
+          child,
+          offset + new Offset(
+            untransformedPaintingCoordinates.dx,
+            // Paint everything in the center (e.g. angle = 0), then transform.
+            -_topScrollMarginExtent,
+          ),
+        );
+      },
+    );
+  }
+
+  /// Apply incoming transformation with the transformation's origin at the
+  /// viewport's center.
+  Matrix4 _centerOriginTransform(Matrix4 originalMatrix) {
+    final Matrix4 result = new Matrix4.identity();
+    final Offset centerOriginTranslation = Alignment.center.alongSize(size);
+    result.translate(centerOriginTranslation.dx, centerOriginTranslation.dy);
+    result.multiply(originalMatrix);
+    result.translate(-centerOriginTranslation.dx, -centerOriginTranslation.dy);
+    return result;
+  }
+
+  @override
+  void applyPaintTransform(RenderBox child, Matrix4 transform) {
+    transform.translate(0.0, _getUntransformedPaintingCoordinateY(0.0));
+  }
+
+  @override
+  Rect describeApproximatePaintClip(RenderObject child) {
+    if (child != null && _shouldClipAtCurrentOffset()) {
+      return Offset.zero & size;
+    }
+    return null;
+  }
+
+  @override
+  bool hitTestChildren(HitTestResult result, { Offset position }) {
+    return false;
+  }
+
+  @override
+  double getOffsetToReveal(RenderObject target, double alignment) {
+    final _ListWheelParentData parentData = target.parentData;
+    final double centerPosition = parentData.offset.dy;
+
+    if (alignment < 0.5) {
+      return centerPosition + _topScrollMarginExtent * alignment * 2.0;
+    } else if (alignment > 0.5) {
+      return centerPosition - _topScrollMarginExtent * (alignment - 0.5) * 2.0;
+    } else {
+      return centerPosition;
+    }
+  }
+
+  @override
+  void showOnScreen([RenderObject child]) {
+    // Shows the child in the selected/center position.
+    offset.jumpTo(getOffsetToReveal(child, 0.5));
```

Comment:
if child is null, this will throw

Reply:
Ah, I didn't read the parent docs correctly. Done.